pandas/io/json/_table_schema.py

```diff
@@ -230,6 +230,10 @@ def build_table_schema(data, index=True, primary_key=None, version=True):
     'pandas_version': '0.20.0',
     'primaryKey': ['idx']}
     """
+
+    if data.index.nlevels == 1 and isinstance(data.index, MultiIndex):
+        data.index = data.index.get_level_values(0)
```

Comment:
Is `data` copied somewhere or is this mutating the frame that the user supplies?

Reply:
Data is indeed modified, thanks.
The simplest solution is to make a copy of `data` in this particular case.
However, a copy of `data` is made after the call of `build_table_schema`.
If we definitely want to avoid the copy, it is possible to make it by modifying both `build_table_schema`, and replacing index in constructor of `JSONTableWriter` once the copy is created.

Reply:
Let's try that for now. We usually want to avoid extra copies